core/dynamo.c

```diff
@@ -1169,7 +1169,8 @@ dynamo_shared_exit(thread_record_t *toexit /* must ==cur thread for Linux */
     return SUCCESS;
```

Comment:
> Those scenarios would need their own optimizer-specific solution.

Not for this PR but a future one -- We should add that to the docs: though we never made particular docs for static-library DR so we need to do that first.  Maybe we should file an issue on doing that.

Reply:
I have it in my to-do list to write the start of the doc, I'll make a note to add this caveat.